### PR TITLE
Fixed breadcrumb links for packages with build metadata

### PR DESCRIPTION
**Pull request by rocklan**
Fixes breadcrumb links for packages with build metadata:

image

The only question now is if we want to continue to display the build metadata in the link (the +6...) part. I think it's ok, but that's me. 

Addresses #8136

### CODE REVIEW

File: src/NuGetGallery/UrlHelperExtensions.cs [hunk 1]
```diff
@@ -228,14 +228,16 @@ public static string Package(this UrlHelper url, string id, bool relativeUrl = t
             string version,
             bool relativeUrl = true)
         {
+            string normalized = NuGetVersionFormatter.Normalize(version);
```
Comment (joelverhagen) [inline src/NuGetGallery/UrlHelperExtensions.cs:231]: I think `version` can be `null` in some contexts, based on usage, e.g. https://github.com/NuGet/NuGetGallery/blob/b3c174ecc7affadf411d88270a50ed4b5eb1f2a4/src/NuGetGallery/Controllers/JsonApiController.cs#L135

I'd recommend either putting this in an `if` to protect from this case or to add a unit test to verify and "lock-in" the behavior of `Normalize` with a `null` value passed in.

Alternatively, you can update the particular call site to ensure `null` is never passed in but I think this is a desirable, centralized behavior.

**Comment by joelverhagen** [comment]
Would it be possible to get a unit test to cover this case?

File: src/NuGetGallery/UrlHelperExtensions.cs (hunk 1 above)
Comment (rocklan) [reply src/NuGetGallery/UrlHelperExtensions.cs]: Null reference checks? They'll never catch on 😄 . Sure, my mistake, done.

**Comment by rocklan** [comment]
> Would it be possible to get a unit test to cover this case?

I've added a unit test to cover this in the `ThePackageBaseHelperMethod` class. And having done so, I can see how this bug snuck in. There is already a test to cover this exact scenario, inside the `ThePackageHelperMethod` class, but the problem is it calls one of the overloads of `Package`, where a `Package` object is passed through, and this function passes through the `package.NormalizedVersion`. However the code that renders the breadcrumbs doesn't call this method, it calls the one that accepts two strings, and it passes though the raw version instead of the normalised one. I don't think calling normalize() twice will cause any issues.

File: tests/NuGetGallery.Facts/UrlHelperExtensionsFacts.cs [hunk 2]
```diff
@@ -53,6 +77,7 @@ public void UsesNormalizedVersionInUrls()
                 Assert.EndsWith(package.NormalizedVersion, fixedUrl);
             }
 
+
```
Comment (joelverhagen) [inline tests/NuGetGallery.Facts/UrlHelperExtensionsFacts.cs:80]: nit: don't need the extra line

Comment (rocklan) [reply tests/NuGetGallery.Facts/UrlHelperExtensionsFacts.cs]: ok done, fixed

**Comment by joelverhagen** [comment]
Hey, it looks like this test is failing on your branch (per CI):
````
##[error]    NuGetGallery.ApiControllerFacts+TheGetStatsDownloadsAction.VerifyRecentPopularityStatsDownloads [FAIL]

      unexpected content result[3].Gallery
      Expected: True
      Actual:   False
      Stack Trace:
        tests\NuGetGallery.Facts\Controllers\ApiControllerFacts.cs(0,0): at NuGetGallery.ApiControllerFacts.TheGetStatsDownloadsAction.<VerifyRecentPopularityStatsDownloads>d__0.MoveNext()
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
        --- End of stack trace from previous location where exception was thrown ---
           at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
           at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
```

**Comment by rocklan** [comment]
> Hey, it looks like this test is failing on your branch (per CI): NuGetGallery.ApiControllerFacts+TheGetStatsDownloadsAction.VerifyRecentPopularityStatsDownloads [FAIL]

I've committed a fix for this, it was previously checking that the version number on the end of a link was `1.1` but now as they are normalized it should be checking for `1.1.0`

**Comment by joelverhagen** [comment]
Awesome, thanks for this contribution @rocklan! 🥇
